cmd/api/main.go

```diff
@@ -127,45 +140,106 @@ func responseJSON(w http.ResponseWriter, status int, body any) {
 	}
 }
 
-/* Stores the entry as a new book in the database, if there isn't one with the same name yet. */
-func createBook(w http.ResponseWriter, r *http.Request) {
+func validateBookEntry(w http.ResponseWriter, r *http.Request) (valid bool, bookEntry Book) {
 
-	var newBook Book
-	err := json.NewDecoder(r.Body).Decode(&newBook) //Read the Json body and save the entry to newBook
+	err := json.NewDecoder(r.Body).Decode(&bookEntry) //Read the Json body and save the entry to bookEntry
 	if err != nil {
 		log.Println(err)
 		errR := errResponse{
-			Code:    errResponseCreateBookInvalidJSON.Code,
-			Message: errResponseCreateBookInvalidJSON.Message + err.Error(),
+			Code:    errResponseBookEntryInvalidJSON.Code,
+			Message: errResponseBookEntryInvalidJSON.Message + err.Error(),
 		}
 		responseJSON(w, http.StatusBadRequest, errR)
-		return
+		return false, bookEntry
 	}
 
-	filled := filledFields(newBook) //Verify if all entry fields are filled.
+	filled := filledFields(bookEntry) //Verify if all entry fields are filled.
 	if !filled {
-		responseJSON(w, http.StatusBadRequest, errResponseCreateBookBlankFileds)
+		responseJSON(w, http.StatusBadRequest, errResponseBookEntryBlankFileds)
+		return false, bookEntry
+	}
+
+	return true, bookEntry
+}
```

Comment:
I get what you are trying here, but I don't think this approach is good.
Lets have these two responsibilities separated, one function to validate the data and another one to render the error in the response.
Our errors already carry a lot of detail, you can use them alone to render the correct error messages.